File: WWF/WF_WCF_Samples v4.0/WCF/Basic/Binding/WS/FederationHttp/CS/Shared/ICalculator.cs
﻿//-----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//-----------------------------------------------------------------------------

using System.Security.Permissions;

using System.ServiceModel;


namespace Microsoft.Samples.WSFederationHttpBinding
{
    // Define a service contract.
    [ServiceContract(Namespace = "http://Microsoft.Samples.WSFederationHttpBinding")]
    public interface ICalculator
    {
        [OperationContract]
        double Add(double n1, double n2);
        [OperationContract]
        double Subtract(double n1, double n2);
        [OperationContract]
        double Multiply(double n1, double n2);
        [OperationContract]
        double Divide(double n1, double n2);
    }
}

File: WWF/WF_WCF_Samples v4.0/WCF/Basic/Binding/WS/FederationHttp/VB/Shared/ICalculator.vb
﻿' Copyright (c) Microsoft Corporation.  All Rights Reserved.

Imports System.Security.Permissions
Imports System.ServiceModel

Namespace Microsoft.Samples.WSFederationHttpBinding

    ' Define a service contract.
    <ServiceContract([Namespace]:="http://Microsoft.Samples.WSFederationHttpBinding")> _
    Public Interface ICalculator

        <OperationContract()> _
        Function Add(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Subtract(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double

    End Interface

End Namespace

